src/index.js

```diff
@@ -122,7 +128,15 @@ export function useVirtual({
     }
 
     return virtualItems
-  }, [range.start, range.end, measurements, sizeKey, defaultScrollToFn])
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [
+    range.start,
+    range.end,
+    measurements,
+    sizeKey,
+    defaultScrollToFn,
+    requestMeasure /* required */,
```

Comment:
requestMeasure should be added to the effect that calls calculateRange, not here.

Reply:
I have updated the PR to use `setMeasuredCache({})` instead of depending on state.